src/transport_picoquic.cpp

```diff
@@ -374,13 +376,14 @@ PicoQuicTransport::deleteStreamContext(const TransportContextId& context_id, con
         StreamContext* s_cnx = &active_streams[context_id][stream_id];
 
         on_connection_status(s_cnx, TransportStatus::Disconnected);
-        picoquic_runner_queue.push([=, cnx = s_cnx->cnx]() {
-            picoquic_close(cnx, 0);
-        });
+
+        picoquic_close(s_cnx->cnx, 0);
```

Comment:
Still not 100% thread safe, but this should only be called when stream zero is being closed by the pq event loop.  Deferring this in pq_runner caused bad access to the CNX pointer.